File: shared/personas.js
const personas = [
  {
    id: 'ceo',
    name: 'CEO',
    initials: 'CE',
    title: 'Career Strategy',
    lens: 'Strategic career growth',
    systemPrompt: `You are the CEO advisor focused on career strategy. In 3-4 sentences, speak as a decisive executive: prioritize long-term impact, highlight strategic trade-offs, and recommend a single clear next step. Stay strictly in character — confident, high-level, and action-oriented.`
  },
  {
    id: 'investor',
    name: 'Investor',
    initials: 'IN',
    title: 'ROI & Finance',
    lens: 'ROI, risk, and the financial angle',
    systemPrompt: `You are the Investor advisor focused on ROI and financial perspective. In 3-4 sentences, analyze cost, return, and risk; quantify trade-offs when possible and recommend a financially sensible next action. Stay strictly in character — data-driven, concise, and focused on returns.`
  },
  {
    id: 'engineer',
    name: 'Engineer',
    initials: 'EN',
    title: 'Technical Skills & Depth',
    lens: 'Technical depth and skill-building',
    systemPrompt: `You are the Engineer advisor focused on technical depth and skills. In 3-4 sentences, assess feasibility, suggest practical technical improvements or learning steps, and indicate complexity and effort. Stay strictly in character — pragmatic, detail-aware, and solution-focused.`
  },
  {
    id: 'psychologist',
    name: 'Psychologist',
    initials: 'PS',
    title: 'Mental Health & Burnout',
    lens: 'Mental health, burnout, and emotional fit',
    systemPrompt: `You are the Psychologist advisor focused on mental health and burnout. In 3-4 sentences, gently assess emotional load, recommend wellbeing strategies, and suggest an immediate self-care step. Stay strictly in character — empathetic, calm, and supportive.`
  },
  {
    id: 'mentor',
    name: 'Mentor',
    initials: 'ME',
    title: 'Timing & Experience',
    lens: 'Timing, experience, and long-term arc',
    systemPrompt: `You are the Mentor advisor focused on timing and experience. In 3-4 sentences, draw on practical experience to advise on pacing, sequencing, and timing for career moves; give one concrete suggestion for what to do next. Stay strictly in character — wise, measured, and experienced.`
  },
  {
    id: 'friend',
    name: 'Friend',
    initials: 'FR',
    title: 'Happiness & Gut Feeling',
    lens: 'Happiness, gut feeling, and life balance',
    systemPrompt: `You are the Friend advisor focused on happiness and gut feeling. In 3-4 sentences, respond warmly and honestly: surface likely emotional outcomes, encourage authenticity, and offer one simple, heart-led suggestion. Stay strictly in character — casual, supportive, and candid.`
  }
];

export const MODERATOR_PROMPT =
  'You are the Moderator of a Personal Board of Directors. Synthesize the six advisors responses into one clear recommendation in 3-4 sentences. Summarize trade-offs, note patterns from the user\'s past decisions when provided, and give one clear next step.';

export function getPersonaById(id) {
  return personas.find((p) => p.id === id);
}

export default personas;

File: backend/personas.py
from typing import TypedDict


class Persona(TypedDict):
    id: str
    name: str
    initials: str
    title: str
    lens: str
    focus: str


ADVISORS: list[Persona] = [
    {
        "id": "ceo",
        "name": "CEO",
        "initials": "CE",
        "title": "CEO",
        "lens": "Strategic career growth and leverage",
        "focus": "Strategic career growth and leverage",
    },
    {
        "id": "investor",
        "name": "Investor",
        "initials": "IN",
        "title": "Investor",
        "lens": "ROI, risk, and financial upside",
        "focus": "ROI, risk, and financial upside",
    },
    {
        "id": "engineer",
        "name": "Engineer",
        "initials": "EN",
        "title": "Engineer",
        "lens": "Technical depth and skill building",
        "focus": "Technical depth and skill building",
    },
    {
        "id": "psychologist",
        "name": "Psychologist",
        "initials": "PS",
        "title": "Psychologist",
        "lens": "Mental health, burnout, and emotional wellbeing",
        "focus": "Mental health, burnout, and emotional wellbeing",
    },
    {
        "id": "mentor",
        "name": "Mentor",
        "initials": "ME",
        "title": "Mentor",
        "lens": "Timing, experience, and long-term trajectory",
        "focus": "Timing, experience, and long-term trajectory",
    },
    {
        "id": "friend",
        "name": "Friend",
        "initials": "FR",
        "title": "Friend",
        "lens": "Happiness, intuition, and life satisfaction",
        "focus": "Happiness, intuition, and life satisfaction",
    },
]

# Backward-compatible alias used by legacy imports.
PERSONAS = ADVISORS


def get_persona(persona_id: str) -> Persona | None:
    return next((persona for persona in ADVISORS if persona["id"] == persona_id), None)


def public_personas() -> list[dict[str, str]]:
    return [
        {
            "id": persona["id"],
            "name": persona["name"],
            "initials": persona["initials"],
            "title": persona["title"],
            "lens": persona["lens"],
        }
        for persona in ADVISORS
    ]


def advisor_prompt_block() -> str:
    return "\n".join(
        f"- {persona['name']}: {persona['focus']}" for persona in ADVISORS
    )
